libdnf/advisory/advisory.cpp

```diff
@@ -0,0 +1,165 @@
+    }
+    if (!strcmp(type, "bugfix")) {
+        return Advisory::Type::BUGFIX;
+    }
+    if (!strcmp(type, "enhancement")) {
+        return Advisory::Type::ENHANCEMENT;
+    }
+    if (!strcmp(type, "security")) {
+        return Advisory::Type::SECURITY;
+    }
+    if (!strcmp(type, "newpackage")) {
+        return Advisory::Type::NEWPACKAGE;
+    }
+
+    return Advisory::Type::UNKNOWN;
+}
+
+const char * Advisory::advisory_type_to_cstring(Type type) {
+    switch (type) {
+        case Type::ENHANCEMENT:
+            return "enhancement";
+        case Type::SECURITY:
+            return "security";
+        case Type::NEWPACKAGE:
+            return "newpackage";
+        case Type::BUGFIX:
+            return "bugfix";
+        default:
+            return NULL;
+    }
+}
+
+std::string Advisory::get_severity() const {
+    Pool * pool = sack->p_impl->get_pool();
+
+    //TODO(amatej): should we call SolvPrivate::internalize_libsolv_repo(solvable->repo);
+    //              before pool_lookup_str?
+    //              pool_lookup_str can return nullptr but std::string constructor cannot
+    //              accept nullptr (e.g. id = 0).
+    //              If so do this just once in solv::advisroy_private
+    return std::string(pool_lookup_str(pool, id.id, UPDATE_SEVERITY));
+}
+
+AdvisoryId Advisory::get_id() const {
+    return id;
+}
+
+std::vector<AdvisoryReference> Advisory::get_references(AdvisoryReferenceType ref_type) const {
+    Pool * pool = sack->p_impl->get_pool();
+
+    std::vector<AdvisoryReference> output;
+
+    Dataiterator di;
+    dataiterator_init(&di, pool, 0, id.id, UPDATE_REFERENCE, 0, 0);
+
+    for (int index = 0; dataiterator_step(&di); index++) {
+        dataiterator_setpos(&di);
+        const char * current_type = pool_lookup_str(pool, SOLVID_POS, UPDATE_REFERENCE_TYPE);
+
+        if (((ref_type & AdvisoryReferenceType::CVE) == AdvisoryReferenceType::CVE &&
+             (strcmp(current_type, "cve") == 0)) ||
+            ((ref_type & AdvisoryReferenceType::BUGZILLA) == AdvisoryReferenceType::BUGZILLA &&
+             (strcmp(current_type, "bugzilla") == 0)) ||
+            ((ref_type & AdvisoryReferenceType::VENDOR) == AdvisoryReferenceType::VENDOR &&
+             (strcmp(current_type, "vendor") == 0))) {
+            output.emplace_back(AdvisoryReference(*sack, id, index));
+        }
+    }
+
+    dataiterator_free(&di);
+    return output;
+}
+
+std::vector<AdvisoryCollection> Advisory::get_collections() const {
+    Pool * pool = sack->p_impl->get_pool();
+
+    std::vector<AdvisoryCollection> output;
+
+    Dataiterator di;
+    dataiterator_init(&di, pool, 0, id.id, UPDATE_COLLECTIONLIST, 0, 0);
+
+    for (int index = 0; dataiterator_step(&di); index++) {
+        dataiterator_setpos(&di);
+        output.emplace_back(AdvisoryCollection(*sack, id, index));
+    }
+
+    dataiterator_free(&di);
+
+    return output;
+}
+
+bool Advisory::is_applicable() const {
```

Comment:
(Note for discussion) I am not sure whether we want to keep it. Applicability is related to Collection and we do not need to keep compatibility with dnf-4. Also I suggest that keeping it can end in improper usage of advisories.

Reply:
Added TODO for this, we can remove it later.